### ISSUE TITLE
Using SemanticModel.GetOperation on an IPropertyInitializer's Syntax produces an IAssignmentExpression, not the original IPropertyInitializer

### ISSUE THREAD

**Issue by jinujoseph** [dotnet/roslyn#18710]
Reported by @evildour Refer dotnet/roslyn-analyzers#1142 (comment)

**Comment by evildour** [comment]
Here is some code which reproduces the issue:
```csharp
var tree = CSharpSyntaxTree.ParseText(@"
using System.Collections.Generic;
class C
{
    void Foo()
    {
        var x = new Dictionary<int, int> { [1] = 2 };
    }
}
",
CSharpParseOptions.Default.WithFeatures(
    new[] { new KeyValuePair<string, string>("IOperation", "true") }));
var compilation = CSharpCompilation.Create(
    "out.dll",
    new[] { tree },
    new[] { MetadataReference.CreateFromFile(typeof(int).Assembly.Location) },
    new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
var semanticModel = compilation.GetSemanticModel(tree);
var objectCreationExpression =
    tree.GetRoot().DescendantNodes().OfType<ObjectCreationExpressionSyntax>().First();
var operation = (IObjectCreationExpression)semanticModel.GetOperation(objectCreationExpression);
foreach (var initializer in operation.MemberInitializers)
{
    var otherOperation = semanticModel.GetOperation(initializer.Syntax);
    if (otherOperation != initializer)
    {
        // initializer is an IPropertyInitializer but otherOperation is an IAssignmentExpression.
    }
}
```

**Comment by mavasani** [comment]
Fixed with #20689. Now we are consistent and do not synthesize IPropertyInitializer. We will always return an IAssignmentExpression whose left is IPropertyReferenceExpression whose Arguments include the indices.
